From 4107c932b0a0de93c7dfb7fc75c14125ea004d79 Mon Sep 17 00:00:00 2001
From: Shivani Agarwal <shivania2@vmware.com>
Date: Fri, 9 Sep 2022 14:20:47 +0000
Subject: [PATCH] Fix bin denials for K8's deployment with containerd (rebased
 for 43.6)

Changes are added to resolve following bin denials

avc:  denied  { watch } for pid=2071 comm="kube-controller"
path="/usr/libexec/kubernetes/kubelet-plugins/volume/exec"
dev="sda3" ino=1458924 scontext=system_u:system_r:container_t:s0:c610,c650
tcontext=system_u:object_r:bin_t:s0 tclass=dir permissive=1
---
 policy/modules/contrib/container.if | 24 ++++++++++++++++++++++++
 policy/modules/contrib/container.te |  1 +
 2 files changed, 25 insertions(+)

diff --git a/policy/modules/contrib/container.if b/policy/modules/contrib/container.if
--- a/policy/modules/contrib/container.if
+++ b/policy/modules/contrib/container.if
@@ -226,6 +226,30 @@
 
 ########################################
 ## <summary>
+##      Allow process to watch bin files.
+## </summary>
+## <desc>
+##      <p>
+##      Allow the specified domain to watch bin files
+##      </p>
+## </desc>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`container_runtime_watch_bin_files',`
+        gen_require(`
+                type bin_t;
+        ')
+
+         allow $1 bin_t:dir watch_dir_perms;
+         allow $1 bin_t:file watch_file_perms;
+')
+
+########################################
+## <summary>
 ##	Manage container share files.
 ## </summary>
 ## <param name="domain">
diff --git a/policy/modules/contrib/container.te b/policy/modules/contrib/container.te
--- a/policy/modules/contrib/container.te
+++ b/policy/modules/contrib/container.te
@@ -1705,3 +1705,4 @@
 files_pid_filetrans(container_runtime_t, system_conf_t, dir, "sysctl.d")
 
 container_runtime_read_kubernetes_files(container_t)
+container_runtime_watch_bin_files(container_t)
